src/adapter/src/catalog/state.rs

```diff
@@ -986,7 +988,10 @@ impl CatalogState {
                         let raw_expr = view.expr;
                         let optimized_expr = optimizer.optimize(raw_expr)?;
 
-                        lock.set(optimized_expr).expect("must set");
+                        desc_lock
```

Comment:
I don't see a reason why the result of

```
let optimized_expr = optimizer.optimize(raw_expr)?;
````

cannot contain both the `optimized_expr` and it's `_.typ()`. Would this help?